### PR TITLE
fix WorkflowInstanceTokenTest

### LINKED ISSUE #2270: WorkflowInstanceTokenTest#shouldCompleteInstanceAfterInterruptingBoundaryEventTriggered is flaky
```
15:53:36 [1 - Java Tests] [ERROR] Tests run: 14, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.906 s <<< FAILURE! - in io.zeebe.broker.workflow.WorkflowInstanceTokenTest
15:53:36 [1 - Java Tests] [ERROR] shouldCompleteInstanceAfterInterruptingBoundaryEventTriggered(io.zeebe.broker.workflow.WorkflowInstanceTokenTest)  Time elapsed: 5.041 s  <<< ERROR!
15:53:36 [1 - Java Tests] io.zeebe.test.util.stream.StreamWrapperException: No event found matching the criteria
15:53:36 [1 - Java Tests] 	at java.util.Optional.orElseThrow(Optional.java:290)
15:53:36 [1 - Java Tests] 	at io.zeebe.test.util.stream.StreamWrapper.getFirst(StreamWrapper.java:80)
15:53:36 [1 - Java Tests] 	at io.zeebe.broker.workflow.WorkflowInstanceTokenTest.assertThatWorkflowInstanceCompletedAfter(WorkflowInstanceTokenTest.java:455)
15:53:36 [1 - Java Tests] 	at io.zeebe.broker.workflow.WorkflowInstanceTokenTest.shouldCompleteInstanceAfterInterruptingBoundaryEventTriggered(WorkflowInstanceTokenTest.java:308)
15:53:36 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
15:53:36 [1 - Java Tests] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
15:53:36 [1 - Java Tests] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
15:53:36 [1 - Java Tests] 	at java.lang.reflect.Method.invoke(Method.java:498)
15:53:36 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
15:53:36 [1 - Java Tests] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
15:53:36 [1 - Java Tests] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
15:53:36 [1 - Java Tests] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
15:53:36 [1 - Java Tests] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
15:53:36 [1 - Java Tests] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
15:53:36 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
15:53:36 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
15:53:36 [1 - Java Tests] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
15:53:36 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:53:36 [1 - Java Tests] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
15:53:36 [1 - Java Tests] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
15:53:36 [1 - Java Tests] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
15:53:36 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:128)
15:53:36 [1 - Java Tests] 	at org.junit.runners.Suite.runChild(Suite.java:27)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
15:53:36 [1 - Java Tests] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
15:53:36 [1 - Java Tests] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #2270

### CODE REVIEW

File: broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceTokenTest.java [hunk 1]
```diff
@@ -43,8 +45,8 @@
 
 public class WorkflowInstanceTokenTest {
 
-  private static EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule();
-  private static ClientApiRule apiRule = new ClientApiRule(brokerRule::getClientAddress);
+  private static final EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule();
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceTokenTest.java]:
```suggestion
  private static final EmbeddedBrokerRule BROKER_RULE = new EmbeddedBrokerRule();
```

File: broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceTokenTest.java [hunk 2]
```diff
@@ -43,8 +45,8 @@
...
+  private static final ClientApiRule apiRule = new ClientApiRule(brokerRule::getClientAddress);
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceTokenTest.java]:
```suggestion
  private static final ClientApiRule API_RULE = new ClientApiRule(brokerRule::getClientAddress);
```

File: broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceTokenTest.java [hunk 3]
```diff
@@ -326,8 +332,7 @@ public void shouldCompleteInstanceAfterNonInterruptingBoundaryEventTriggered() {
         testClient.createWorkflowInstance(r -> r.setBpmnProcessId(processId)).getInstanceKey();
 
     // when
-    testClient.receiveElementInState(
-        workflowInstanceKey, "task-1", WorkflowInstanceIntent.ELEMENT_ACTIVATED);
+    testClient.receiveTimerRecords().withIntent(TimerIntent.CREATED).getFirst();
```
Comment (saig0) [inline broker-core/src/test/java/io/zeebe/broker/workflow/WorkflowInstanceTokenTest.java]: In a class rule, there could be multiple timer records. 
```suggestion
   testClient
        .receiveJobs()
        .withWorkflowInstanceKey(workflowInstanceKey)
        .withIntent(JobIntent.CREATED)
        .getFirst();
```

**Review by saig0 — CHANGES_REQUESTED** [review]
Nice fix :+1:

**Review by saig0 — APPROVED** [review]

**Comment by Zelldon** [comment]
bors retry

**Pull request merged**